### PR TITLE
Feature/paginator

### PR DESCRIPTION
**Pull request by chennan47**
resolve issue: #1695

<b>Purpose</b>
solve the problem that there is no pagination for add link search result
image

<b>Changes</b>
add paginator to add link serach result
image

### CODE REVIEW

**Comment by chennan47** [comment]
@sloria this PR itself is rdy, but need the evaluation on ES6 before review and merge.

**Comment by sloria** [comment]
@chennan47 or @samchrisinger Can you please update this PR to use the [oop](https://github.com/CenterForOpenScience/osf.io/blob/9e53712dba3aa7ff33b35808cf2cd0032fc37eb2/website/static/js/oop.js) module in place of ES6 classes?

**Comment by chennan47** [comment]
@sloria already updated PR to use oop module

File: website/static/js/contribAdder.es6.js [hunk 1]
```diff
@@ -5,107 +5,109 @@ var $ = require('jquery');
 var ko = require('knockout');
 var $osf = require('osfHelpers');
 var bootbox = require('bootbox');
+var Paginator = require('./paginator');
```
Comment (sloria) [inline website/static/js/contribAdder.es6.js]: Use `js/paginator`

File: website/static/js/paginator.es6.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
+var ko = require('knockout');
```
Comment (sloria) [inline website/static/js/paginator.es6.js]: Missing `'use strict';`

File: website/static/js/paginator.es6.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+var oop = require('js/oop');
+
+var MAX_PAGES_ON_PAGINATOR = 7;
+var MAX_PAGES_ON_PAGINATOR_SIDE = 5;
```
Comment (sloria) [inline website/static/js/paginator.es6.js]: Trim trailing whitespace.

File: website/static/js/paginator.es6.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+var MAX_PAGES_ON_PAGINATOR = 7;
+var MAX_PAGES_ON_PAGINATOR_SIDE = 5;    
+
+var Paginator = oop.defclass({
+    constructor() {
+        this.numberOfPages = ko.observable(0);
+        this.currentPage = ko.observable(0);
+        this.paginators = ko.observableArray([]);
+        if (!this.search){
```
Comment (sloria) [inline website/static/js/paginator.es6.js]: @samchrisinger @chennan47 This doesn't quite work. The constructor may be executed before the class's `search()` method is defined. How about this instead:

```js
var Paginator = oop.defclass({
    // ...
    search() {
        throw new Error('Paginator subclass must define a "search" method.');
    }
});

**Comment by sloria** [comment]
Pass finished. :v:

**Comment by sloria** [comment]
Also, tests should be added for `Paginator`. This can be a follow-up PR

File: website/static/js/paginator.es6.js (hunk 3 above)
Comment (chennan47) [inline website/static/js/paginator.es6.js]: solved in 678620f

File: website/static/js/paginator.es6.js (hunk 4 above)
Comment (chennan47) [inline website/static/js/paginator.es6.js]: solved in 678620f

File: website/static/js/paginator.es6.js (hunk 2 above)
Comment (chennan47) [inline website/static/js/paginator.es6.js]: solved in 678620f

**Comment by chennan47** [comment]
@sloria rdy for another pass

**Pull request merged**